src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -108,37 +108,57 @@ public dynamic LoadDocument(object keyOrEnumerable, string collectionName)
                 keyOrEnumerable.GetType().FullName + ": " + keyOrEnumerable);
         }
 
+        protected IEnumerable<AbstractField> CreateField(string name, object value, CreateFieldOptions options)
+        {
+            // IMPORTANT: Do not delete this method, it is used by the indexes code when using CreateField
+
+            options = options ?? CreateFieldOptions.Default;
+
+            IndexFieldOptions allFields = null;
+            var scope = CurrentIndexingScope.Current;
+            if (scope.IndexDefinition is MapIndexDefinition mapIndexDefinition)
```

Comment:
what about map /reduce indexes?

Reply:
JS indexes?

Reply:
MapReduceIndexDefinition inherits from MapIndexDefinition so it is fine.

JS indexes do not support 'CreateField' at all

Reply:
You can certainly do dynamic fields in JS, see:

```
 @"map('Users', function (u){
                                    return {
                                        Name: u.Name,
                                        _: {$value: u.Name, $name:'AnalyzedName', $options:{index: true, store: true}}
                                    };
                                })",
```

Reply:
I will check that tomorrow

Reply:
Done. Changed syntax a bit to cover more options.

Reply:
'storage' supports both:
- true/false
- 'Yes' / 'No'